vm.go

```diff
@@ -306,12 +306,29 @@ type vm struct {
 	interrupted   uint32
 	interruptVal  interface{}
 	interruptLock sync.Mutex
+
+	lastFunctionTicks uint64
 }
 
 type instruction interface {
 	exec(*vm)
 }
 
+// profileTicks tracks the ticks for the current Program, this should be called prior to updating the program (i.e. vm.prg = p)
+func (vm *vm) profileTicks() {
+	if vm.r.functionTickTrackingEnabled {
+		currentTicks := vm.r.Ticks()
+		if vm.prg != nil {
+			function := string(vm.prg.funcName)
+			if vm.prg.src != nil {
+				function = vm.prg.src.Name() + "_" + function
```

Comment:
[mega-nit] function names could have an underscore. Thoughts on going with a period instead?
```suggestion
				function = vm.prg.src.Name() + "." + function
```

Reply:
The only reason I went with a _ is because the prg.src.Name() can be a file name so `test.js.functionName` looked a bit weird. I can go back to the period though, I don't have a strong opinion on this.